### ISSUE TITLE
Support `.nil?` in DSL

### ISSUE THREAD

**Issue by pelletencate** [rzane/baby_squeel#93]
Right now, in order to check if something IS NULL, I have to use `field == nil` in the DSL. Wouldn't it be more ruby-esque to be able to say `field.nil?` instead?

**Comment by rzane** [comment]
Hmm... sort of, but this feels pretty like it would cause unexpected problems that would be really hard to figure out what went wrong. Plus, then we'd have to support `!field.nil?` to produce an `IS NOT NULL` query. 

I would rather add `is_null(field)` and `is_not_null(field)`

**Comment by chewi** [comment]
One problem with `field == nil` is that RuboCop keeps trying to replace it with `.nil?` !

**Comment by pelletencate** [comment]
Yeah, the initial reason why I proposed this change was Rubocop complaining. I feel kinda stupid not thinking about `!field.nil?`, that's a more legit concern than whatever I have.

tbh, `is_null` or `is_not_null`, while it would please rubocop, would in my opinion be a step in the wrong direction. A better thing would be to have @bbatsov et al join in, to think about the possibility to provide a custom rubocop plug with your gem, or – more generally – if there's anything that would help take care of DSL that overloads the `==` and operator

As far as I know, the two issues rubocop can complain about are:

* `== nil` and `== 0`
* multiline `{ }` blocks. Although, a solution is to use `do` and `end`, unless you chain (e.g. ` }.limit(5).sum(:amount)`), in which case Rubocop actually has a config that allows you to ignore it.

**Comment by bbatsov** [comment]
> One problem with field == nil is that RuboCop keeps trying to replace it with .nil? !

That's configurable. :) Obviously a static analyzer can't really figure out that something might be a special DSL. I think in such situations you can just exclude the DSL files from the checks and be done with it.
